### ISSUE TITLE
SC2034 false positive when function changes -- but doesn't use -- a global variable

### ISSUE THREAD

**Issue by russellshackleford** [koalaman/shellcheck#3261]
#### For bugs with existing features

- Rule Id: SC2034
- My shellcheck version: 0.10.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2034)
- [ ] ~I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit~ (doesn't seem to support multiple scripts)

#### Here's a snippet or screenshot that shows the problem:

main.sh
```sh
#!/usr/bin/env bash
set -ETeou pipefail
source "$(dirname "$0")"/functions.sh
declare -g globalvar
test_func
echo "$globalvar"
```

functions.sh
```sh
#!/usr/bin/env bash

function test_func() {
  globalvar="something"
}
```

#### Here's what shellcheck currently says:
```
In functions.sh line 4:
  globalvar="something"
  ^-------^ SC2034 (warning): globalvar appears unused. Verify use (or export if used externally).
```

#### Here's what I wanted or expected to see:
No error from shellcheck, bash runs this script fine. `echo "$globalvar"` prints `something`

#### Additional

Doesn't matter if I run shellcheck on just functions.sh or both in the same invocation. Nor does it matter if I use -x or not.
